mm2src/common/log.rs

```diff
@@ -764,31 +759,21 @@ impl LogState {
 
     /// Read-only access to the status dashboard.
     pub fn with_dashboard(&self, cb: &mut dyn FnMut(&[Arc<Status>])) {
-        let dashboard = self.dashboard.spinlock(77).unwrap();
+        let dashboard = self.dashboard.lock();
         cb(&dashboard[..])
     }
 
     pub fn with_tail(&self, cb: &mut dyn FnMut(&VecDeque<LogEntry>)) {
-        match self.tail.spinlock(77) {
-            Ok(tail) => cb(&*tail),
-            Err(_err) => writeln("with_tail] !spinlock"),
-        }
+        let tail = self.tail.lock();
+        cb(&*tail)
     }
 
     pub fn with_gravity_tail(&self, cb: &mut dyn FnMut(&VecDeque<String>)) {
-        let gravity = match self.gravity.spinlock(77) {
-            Ok(guard) => guard,
-            Err(_err) => {
-                writeln("with_gravity_tail] !spinlock");
-                return;
-            },
-        };
-        if let Some(ref gravity) = *gravity {
+        let gravity = self.gravity.lock().clone();
+        if let Some(gravity) = gravity {
```

Comment:
`if let Some(gravity) = self.gravity.lock().clone() {` can be used straight away

Reply:
Done